### DISCUSSION TITLE
Transformation support for existing gain maps: crop, resize, compress

### DISCUSSION THREAD

**Discussion by gregbenz** [google/libultrahdr#254] [Q&A] [answered]
Is support for simple transformations of an existing gain map (crop, resize, re-compress) planned for ultrahdr_app? Or is this only intended for the library and therefore an additional tool like ImageMagick would likely be required?

If this is already possible in [ImageMagick](https://github.com/ImageMagick/ImageMagick/issues/6377), could someone share some examples of the command sequence you could use to reprocess an existing gain map? This will of course be helpful for applications such as media libraries or other uploads to websites (derivative sizes, thumbnails, square crops for a grid, etc).

**Comment by ram-mohan** [comment, marked as answer]
In ultrahdr_api.h there are few editor functions. uhdr_add_effect_mirrror, uhdr_add_effect_rotate, uhdr_add_effect_crop and uhdr_add_effect_resize. On the encoding side, if these functions are configured, these effects are first applied on hdr intent and sdr intent and then encoding is done. On the decoding side, if these functions are configured, then these effects are applied on decoded output and decoded gainmap image. uhdr_get_decoded_image(), uhdr_get_gain_map_image() can be used to get these results. 

The library as of now does not handle applying these effects on a ultrahdr image and give out a recompressed ultrahdr image with effects applied.

**Comment by gregbenz** [comment]
@ram-mohan Thank you, helpful to know the current API.
